x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -359,34 +362,44 @@ private MethodSpec addRawInput() {
             builder.addStatement("return");
         }
         builder.endControlFlow();
+        builder.addStatement("$T block = page.getBlock(channels.get(0))", valueBlockType(init, combine));
+        builder.addStatement("$T vector = block.asVector()", valueVectorType(init, combine));
+        if (includeTimestampVector) {
+            builder.addStatement("$T timestampsBlock = page.getBlock(channels.get(1))", LONG_BLOCK);
+            builder.addStatement("$T timestampsVector = timestampsBlock.asVector()", LONG_VECTOR);
+            builder.beginControlFlow("if (timestampsVector == null) ");
+            builder.addStatement("throw new IllegalStateException($S)", "expected @timestamp vector; but got a block");
+            builder.endControlFlow();
```

Comment:
I think this is fine. But it's worth saying out loud: Things like this *usually* should become `null` and a warning. They'd put the agg in an "I'm broken" state and produce a warning on output. Like `sum` should do if it overflows. It doesn't do that now, but it should.

Anyway, I think this is fine to just hard fail here - just like this - because we're going to want to build machinery around the agg to make sure that it's input is a time series. Which will have the constraint that the `timestamp` is always single valued. And, probably, *descending*.